src/server/routes/emailTemplates.ts

```diff
@@ -0,0 +1,65 @@
+import { Request, Router } from 'express';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { render } from 'mjml-react';
+
+import { ResetPassword } from '../../email/templates/ResetPassword/ResetPassword';
+import { ResetPasswordText } from '../../email/templates/ResetPassword/ResetPasswordText';
+import { Verify } from '../../email/templates/Verify/Verify';
+import { VerifyText } from '../../email/templates/Verify/VerifyText';
+import { AccidentalEmail } from '../../email/templates/AccidentalEmail/AccidentalEmail';
+import { AccidentalEmailText } from '../../email/templates/AccidentalEmail/AccidentalEmailText';
+
+const router = Router();
+
+const emailTemplateTypes = [
+  'reset-password',
+  'verify',
+  'accidental-email',
+] as const;
+type EmailTemplateType = typeof emailTemplateTypes[number];
+
+type EmailRenderResult = {
+  plain: string;
+  html: string;
+};
+
+const renderEmailTemplate = (
+  template: EmailTemplateType,
+): EmailRenderResult | undefined => {
+  switch (template) {
+    case 'reset-password':
+      return {
+        plain: ResetPasswordText(),
+        html: render(ResetPassword()).html,
+      } as EmailRenderResult;
+    case 'verify':
+      return {
+        plain: VerifyText(),
+        html: render(Verify()).html,
+      } as EmailRenderResult;
+    case 'accidental-email':
+      return {
+        plain: AccidentalEmailText(),
+        html: render(AccidentalEmail()).html,
+      } as EmailRenderResult;
+    default:
+      // We don't want to do anything for invalid template names
+      return undefined;
+  }
+};
+
+/* GET a valid email template name.
+ * Returns 406 for invalid template names. */
+router.get(
+  '/email/:template',
+  (req: Request, res: ResponseWithRequestState) => {
+    const template = req.params.template as EmailTemplateType;
+    const templateIsValid = emailTemplateTypes.includes(template);
+
+    return templateIsValid
+      ? res.json(renderEmailTemplate(template))
+      : res.send(406); // 406: Not Acceptable
```

Comment:
I'd suggest `400` over `406` here.

As according to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406) `406`  client error response code indicates that the server cannot produce a response matching the list of acceptable values defined in the request's proactive [content negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation) headers. In this case we're not relying on those headers, but rather malformed client request, which is what [`400`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) is used for.

However I think `404` makes the most sense, if an template doesn't exist, then it should be a "Not Found" case is for, as the request itself isn't malformed in any obvious way.

I also suggest [`res.sendStatus`](https://expressjs.com/en/4x/api.html#res.sendStatus) or [`res.status`](https://expressjs.com/en/4x/api.html#res.status) if you just need to return a status code, as (i think) `res.send(406)` will literally just send 406 in the body, with an okay response, but I'm not sure about that.

```suggestion
      : res.sendStatus(404);
```